B-Sockets-ServerTesting/sockets-servertesting.md

```diff
@@ -0,0 +1,286 @@
+Sockets and Server Testing
+===========================
+Sockets and HTTP
+---------------------------
+### What is a socket?
+
+A socket is a software construct used for many modes of communication between
+processes. The mode of communication that this recitation will focus on is
+network communication. In particular, stream sockets represent an endpoint for
+reliable, bidirectional connections such as TCP connections. This allows for two
+processes, on separate computers, to communicate over a TCP/IP network
+connection.
+
+Sockets have:
+- an IP address, to (typically) identify the computer that the socket endpoint
+  belongs to
+- a port number, to identify which process running on the computer the socket
+  endpoint belongs to
+- a protocol, such as TCP (reliable) or UDP (unreliable). Stream sockets use
+  TCP.
+
+An IP address and port number are both required in order for a computer to
+communicate with a specific process on a remote computer. Think about would
+happen if a port number wasn't necessary.
```

Comment:
Should this be something more along the lines of "Think about what would happen if we didn't have port numbers"? Not quite sure what this is getting at lol